### PR TITLE
configuration interface & commands

### PR DESCRIPTION
**Pull request by cryi**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ node_modules/
 config.json
 config.hjson
 .env
+yarn.lock
```
Comment (kalouo) [inline .gitignore:5]: Should we be adding the `package-lock.json` file here too?

It also makes me think that enforcing the Node version could be useful.

File: src/cli/cmd/pay.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+} from "src/tezos-client";
+import { arePaymentsRequirementsMet } from "src/engine/validate";
+import { cliOptions } from "src/cli";
+
+export const pay = async (cycle: number) => {
+  if (cliOptions.dryRun) {
+    console.log(`Running in 'dry-run' mode...`);
+  }
+
+  const config = get_config();
```
Comment (kalouo) [inline src/cli/cmd/pay.ts:20]: camel case here?

File: test/helpers/index.ts [hunk 3]
```diff
@@ -15,16 +15,22 @@ export const generateConfig = (
     fee_income_recipients?: { [key: string]: string };
     bond_reward_recipients?: { [key: string]: string };
   } = {}
-): Config => {
+): BreadcrumbsConfiguration => {
   return {
     baking_address: args.baking_address || DEFAULT_BAKER,
-    default_fee: args.default_fee || "5",
+    default_fee: !isNaN(Number(args.default_fee))
```
Comment (kalouo) [inline test/helpers/index.ts:21]: I'd say the bang is adding an unnecessary cognitive load. Suggest: 

```javascript
    default_fee: isNaN(Number(args.default_fee))?
      ? 0
      :  Number(args.minimum_payment_amount),
```

Comment (kalouo) [reply test/helpers/index.ts:21]: Same for ones below.

File: .gitignore (hunk 1 above)
Comment (cryi) [reply .gitignore:5]: You usually want some lock file. But it is likely that these two wont be in sync so better to keep yours. 

Enforcing node version is not common and node tends to be backward compatible so I do not think we should do that

File: test/helpers/index.ts (hunk 3 above)
Comment (cryi) [reply test/helpers/index.ts]: fixed

File: .gitignore (hunk 1 above)
Comment (cryi) [reply .gitignore:5]: fixed

File: src/cli/cmd/pay.ts (hunk 2 above)
Comment (cryi) [reply src/cli/cmd/pay.ts]: fixed

File: package.json [hunk 4]
```diff
@@ -4,7 +4,7 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "pay": "ts-node index.ts",
+    "pay": "ts-node index.ts pay",
```
Comment (kalouo) [inline package.json:7]: Trying to run this locally as `npm run pay --cycle=489 --dry-run` but it's no longer recognizing the arguments. Does it have to be run differently?

File: src/cli/index.ts [hunk 5]
```diff
@@ -1,29 +1,28 @@
...
+    .action(pay);
+
+  // we need to set global options before action is executed
+  program.hook("preAction", () => {
+    cliOptions = program.opts();
+  });
+  await program.parseAsync();
+};
+
+export const get_cli_option = (opt: keyof OptionValues) => {
```
Comment (kalouo) [inline src/cli/index.ts:26]: Camel case 🙏

File: package.json (hunk 4 above)
Comment (cryi) [reply package.json:7]: Yea it should be now `npm run pay -- --dry-run 489`

**Review by kalouo — APPROVED** [review]

**Pull request merged**